okhttp-sse/src/main/java/okhttp3/internal/sse/RealEventSource.java

```diff
@@ -54,6 +55,17 @@ public void connect(OkHttpClient client) {
     BufferedSource source = response.body().source();
     ServerSentEventReader reader = new ServerSentEventReader(source, this);
 
+    if (!response.isSuccessful()) {
+      listener.onFailure(this, null, response);
```

Comment:
You don't have to fix it in this PR, but I notice we never ensure the response body is closed in normal SSE operation or these new failure callbacks. We'll need a few try/finallys, but that's an easy follow-up.